snapcraft/internal/repo/_deb.py

```diff
@@ -186,18 +188,74 @@ def get_packages_for_source_type(cls, source_type):
         return packages
 
     @classmethod
-    def install_build_packages(cls, package_names):
+    def _setup_foreign_arch(cls, arch):
+        native_arch = subprocess.check_output(
+            ['dpkg', '--print-architecture']).decode(
+                sys.getfilesystemencoding())
+        if arch in native_arch:
+            return
+        foreign_archs = subprocess.check_output(
+            ['dpkg', '--print-foreign-architectures']).decode(
+                sys.getfilesystemencoding())
+        if arch in foreign_archs:
+            return
+        logger.info('Adding foreign architecture {}'.format(arch))
+        subprocess.check_output(
+            ['sudo', 'dpkg', '--add-architecture', arch])
+
+    @classmethod
+    def _setup_multi_arch_sources(cls, apt_cache, package_name):
+        """If the given package has an arch suffix:
+        Generate arch-specific sources list
+        Update the package cache to pull in the new packages
+        Verify that the package is in the cache
```

Comment:
It's not completely clear from the function name or the docstring what the return value indicates.